v1/contracts/Voting.sol

```diff
@@ -33,9 +99,109 @@ contract Voting {
      * committer can reveal their vote.
      */
     function revealVote(bytes32 identifier, uint time, int price, int salt) external {
-        VoteInstance storage voteInstance = requests[identifier][time];
+        VoteInstance storage voteInstance = priceResolutions[identifier][time].votes[currentRoundNumber];
         require(keccak256(abi.encode(price, salt)) == voteInstance.committedHashes[msg.sender],
                 "Committed hash doesn't match revealed price and salt");
         delete voteInstance.committedHashes[msg.sender];
     }
+
+    /**
+     * @notice Enqueues a request (if a request isn't already present) for the given `identifier`, `time` pair.
+     * @dev Returns the time at which the user should expect the price to be resolved.
+     */
+    function requestPrice(bytes32 identifier, uint time) external returns (uint expectedTime) {
+        uint priceResolutionRound = priceResolutions[identifier][time].lastVotingRound;
+        uint secondsInWeek = 60*60*24*7;
+
+        if (priceResolutionRound == 0) {
+            // Price has never been requested.
+            // Add price request to the next round.
+            _addPriceRequestToRound(_calcVotingRound().add(1), PriceRequest(identifier, time));
```

Comment:
If we want to support the first type of behavior, price requests are guaranteed to be enqueued in the next voting round. If that's the case, it may be unnecessary to index price requests by the voting round. Instead, we can just track `currentRequests` and `upcomingRequests`, as well as the start time for the next voting round.

Reply:
When I wrote this, I was assuming we would disallow future price requests. For now, I'll put a check in to revert on a future request. We should probably come to a more concrete decision about this, though.

If we want to allow future price requests, it would mean that we need to have deterministic voting round numbers for particular time periods. It would also mean that we would need to be able to pre-compute the voting round when requests are received. I'll put a TODO in the code to figure this out.

Reply:
As discussed IRL, the indexing by voting round is to allow users to easily compute the rewards that they are owed.

Reply:
I added a TODO and a check for this.